src/python/grpcio_tests/tests/unit/framework/interfaces/face/_future_invocation_asynchronous_event_service.py

```diff
@@ -450,6 +452,7 @@ def testFailedUnaryRequestUnaryResponse(self):
           with self.assertRaises(face.ExpirationError):
             response_future.result()
           self.assertIsNotNone(response_future.traceback())
+          self.assertTrue(abortion_callback._called)
```

Comment:
My mistake, you are correct.  It was hanging because of a bug in ```add_cancellation_callback``` (The lambda should have not accepted any args, and was failing quietly).  I've made the change.